lib/parse.js

```diff
@@ -121,7 +122,7 @@ var parseObject = function (chain, val, options, valuesParsed) {
         var root = chain[i];
 
         if (root === '[]' && options.parseArrays) {
-            obj = [].concat(leaf);
+            obj = leaf ? [].concat(leaf) : [];
```

Comment:
what if the leaf is a falsy primitive?

Reply:
changed to `obj = typeof leaf === 'string' && leaf === '' ? [] : [].concat(leaf);`